### ISSUE TITLE
IT Glue JSON Forbidden

### ISSUE THREAD

**Issue by ttmcmurry** [marcusolsson/grafana-json-datasource#3]
Hello!

I figured I'd give this plugin a try and interface with IT Glue to produce some useable dashboards.  However, I'm having a devil of a time with the server-end coming back with a 500 forbidden.

According to their [docs](https://api.itglue.com/developer), all I need is to enter two header values:

Content-Type application/vnd.api+json
x-api-key {{my ITG API KEY}}

...and the main url https://api.itglue.com.  There's not much to it.  There is no basic/cert auth.  I've successfully been using Postman for some time, and those two header values are all that's needed to connect and start building out queries, etc.

Unfortunately the plugin doesn't generate a log in Grafana, and I cannot elaborate more on the transaction between the plugin and IT Glue's server URL.  

What I know is a cookie is sent from the api back to the client - AWSALB={{some random value}}.  It's not possible to predict what the cookie value will be, therefore if this is a requirement of the plugin, I'm unsure how to whitelist that cookie. 

From the Grafana server itself, I have the IT Glue PowerShell API wrapper installed and am able to get data I'm looking for, so it isn't a connectivity or key issue.  I'm using the same API Key in powershell as I am in the Grafana Plugin.

**Plugin**
![image](https://user-images.githubusercontent.com/44306558/94623844-1fa7e400-0283-11eb-9924-c801148f96f1.png)

**Postman**
![image](https://user-images.githubusercontent.com/44306558/94623970-6f86ab00-0283-11eb-84c9-52be20fbac06.png)

**PowerShell on Grafana Host**
![image](https://user-images.githubusercontent.com/44306558/94625627-351f0d00-0287-11eb-8ea6-84e1f950537a.png)

**Comment by marcusolsson** [comment]
I'm sorry you've had to spend the time debugging this, though I'm grateful that you took the time to report it!

If I were to guess, it's indeed related to the cookie returned. I don't see a free tier on their website. Is there any way I could replicate this without having a paid account? Do they have any API docs I could look at?

I'll look around and see if become any wiser!

**Comment by ttmcmurry** [comment]
Thanks Marcus.  Their documentation is at https://api.itglue.com/developer.  The difficult part of this is not having a log from the plugin or Grafana to help indicate what isn't working right.  

That said, perhaps https://github.com/itglue/powershellwrapper will help understand what the PowerShell API Wrapper is doing to and provide some clues as to how the session is handled?  It's really just a guess on my end.  :)

**Comment by mbarradas** [comment]
I had the same type of problem you can see the log on the grafana log it shows the call to the url and the response. In my case the problem seem to be related to the fact that the call is done with the IP of the endpoint instead of the url and my endpoint blocks those types of calls. 
I tried it with https://k1toronto.clubspeedtiming.com/api/ and I was able to connect but had other problems.
 Hope this helps.

**Comment by mbarradas** [comment]
Changed my the of my API to localhost (form my use case this was a valid url) instead of the full url of and I was able to connect to the API with this little change. I can confirm that the only way my API is not reachable is by the IP resolved from the DNS. The result was the same with postman and with the panel.

**Comment by marcusolsson** [comment]
Thanks @mbarradas for the update. It sounds like it's a different problem from what @ttmcmurry describes, since they don't use an IP address as URL? For now, maybe add your details to https://github.com/marcusolsson/grafana-jsonapi-datasource/issues/4 instead.

I'll admit that I'm not sure where to go from here. I'm not aware of any way to make Grafana log these requests. I could try to add more details to the error in the data source test message.

**Comment by ttmcmurry** [comment]
@marcusolsson I'm stuck at the same place as you are.  At this point - I'd have to drop to data packet capture and analysis - or finding a useful Grafana log / way to make Grafana log more info.

**Comment by marcusolsson** [comment]
@ttmcmurry Did you manage to fix/get around this eventually?

**Comment by ttmcmurry** [comment]
I gave it another go about 3 weeks ago, and still couldn't make any progress with it.  That said, it became more time efficient to settle with what BrightGauge could do for us as I had to wrap things up and make some progress.  

If I had more time to dig into what is really happening at the packet level - since getting useable logs is difficult - maybe some clues would be revealed, but the problem, unfortunately, is time.
